packages/core/src/providers/studio/index.ts

```diff
@@ -102,76 +111,55 @@ export class StudioLivepeerProvider extends BaseLivepeerProvider {
         headers: this._defaultHeaders,
       },
     );
-
-    return studioStreamSessions.map((studioStreamSession) =>
-      this._mapToStreamSession(studioStreamSession),
-    );
+    return studioStreamSessions;
   }
 
   async createAsset(args: CreateAssetArgs): Promise<Asset> {
-    const tusOptions: tus.UploadOptions = {};
-
-    const existingUpload = new tus.Upload(args.file, {});
-    const previousUploads = await existingUpload.findPreviousUploads();
-
-    let assetId: string | undefined;
-
-    if (previousUploads.length > 0 && previousUploads[0]) {
-      const previousUpload = previousUploads[0];
-
-      assetId = previousUpload.metadata.id;
-
-      await new Promise<void>((resolve, reject) => {
-        const upload = new tus.Upload(args.file, {
-          ...tusOptions,
-          onError(error) {
-            reject(error);
-          },
-          onSuccess() {
-            return resolve();
-          },
-        });
-
-        upload.resumeFromPreviousUpload(previousUpload);
-        upload.start();
-      });
-    } else {
-      const requestUploadEndpoint = await this._create<
-        { tusEndpoint: string; asset: { id: string } },
-        { name: string }
-      >('/asset/request-upload', {
-        json: {
-          name: args.name,
+    const uploadReq = await this._create<
+      { tusEndpoint: string; asset: { id: string } },
+      Omit<CreateAssetArgs, 'file'>
+    >('/asset/request-upload', {
+      json: {
+        name: args.name,
+        meta: args.meta,
+      },
+      headers: this._defaultHeaders,
+    });
+    const {
+      tusEndpoint,
+      asset: { id: assetId },
+    } = uploadReq;
+
+    await new Promise<void>((resolve, reject) => {
+      const upload = new tus.Upload(args.file, {
+        endpoint: tusEndpoint,
+        metadata: {
+          id: assetId,
         },
-        headers: this._defaultHeaders,
-      });
-
-      assetId = requestUploadEndpoint.asset.id;
+        uploadSize: args.uploadSize,
 
-      await new Promise<void>((resolve, reject) => {
-        const upload = new tus.Upload(args.file, {
-          metadata: {
-            id: requestUploadEndpoint.asset.id,
-          },
-          ...tusOptions,
-          endpoint: requestUploadEndpoint.tusEndpoint,
-
-          onError(error) {
-            reject(error);
-          },
-          onSuccess() {
-            return resolve();
-          },
-        });
-
-        upload.start();
+        onError(error) {
+          reject(error);
+        },
+        onProgress(bytesSent, bytesTotal) {
+          if (args.onUploadProgress) {
+            args.onUploadProgress(bytesSent / bytesTotal);
+          }
+        },
+        onSuccess() {
+          return resolve();
+        },
       });
-    }
-
-    if (!assetId) {
-      throw new Error('Could not find asset ID');
-    }
-
+      upload
+        .findPreviousUploads()
```

Comment:
Yeah! That's why I removed the find previous uploads without any endpoint configured. Would you rather remove this one as well, to avoid misleading readers?

Reply:
I think this is great! Just wanted to flag for future work so we don't forget